addons/sale_stock/models/sale_order.py

```diff
@@ -35,31 +35,27 @@ def _default_warehouse_id(self):
     picking_ids = fields.One2many('stock.picking', 'sale_id', string='Transfers')
     delivery_count = fields.Integer(string='Delivery Orders', compute='_compute_picking_ids')
     procurement_group_id = fields.Many2one('procurement.group', 'Procurement Group', copy=False)
-    effective_date = fields.Date("Effective Date", compute='_compute_effective_date', store=True, help="Completion date of the first delivery order.")
-    expected_date = fields.Datetime( help="Delivery date you can promise to the customer, computed from the minimum lead time of "
-                                          "the order lines in case of Service products. In case of shipping, the shipping policy of "
+    expected_date = fields.Datetime(help="Delivery date you can promise to the customer, computed from the minimum lead time of "
+                                          "the order lines in case of Service products.\n In case of shipping, the shipping policy of "
                                           "the order will be taken into account to either use the minimum or maximum lead time of "
-                                          "the order lines.")
+                                          "the order lines.",
+                                    default=fields.Datetime.now)
+    is_delivered = fields.Boolean(compute='_compute_is_delivered', help="Technical field used to indicates if all of the products in the SO were fully delivered")
 
-    @api.depends('picking_ids.date_done')
-    def _compute_effective_date(self):
-        for order in self:
-            pickings = order.picking_ids.filtered(lambda x: x.state == 'done' and x.location_dest_id.usage == 'customer')
-            dates_list = [date for date in pickings.mapped('date_done') if date]
-            order.effective_date = min(dates_list).date() if dates_list else False
+    def _update_expected_date(self):
```

Comment:
https://www.odoo.com/documentation/12.0/reference/guidelines.html#symbols-and-conventions